main.go

```diff
@@ -34,7 +34,7 @@ func setupSettings(adminPass string, confFile string) error {
 	if err != nil {
 		return fmt.Errorf("unable to load settings: %w", err)
 	}
-	if len(settings.StreamKey) == 0 {
+	if !settings.NewStreamKey || len(settings.StreamKey) == 0 {
```

Comment:
This check is incorrect.  If a stream key is being generated it shouldn't matter if a stream key is set in the settings file.  You'll want the two checks separated by `&&` instead of `||`.

![image](https://github.com/zorchenhimer/MovieNight/assets/6540297/18e2b9b2-9220-41c0-92bb-a8669c1780bb)